src/NBomber/Domain/Scenario.fs

```diff
@@ -74,13 +75,32 @@ module Validation =
             | Ok _ -> Ok scenario
             | Error errors -> errors |> List.head |> AppError.createResult
 
+    let checkEmptyOrDuplicatedThresholds (scenario: Contracts.Scenario) =
+        scenario.Thresholds
+        |> Option.map (fun thresholds ->
+            match thresholds with
+            | [] -> AppError.createResult(EmptyThresholds scenario.ScenarioName)
+            | _ ->
+                thresholds
+                |> List.groupBy (fun x -> x.Name)
+                |> List.choose (fun (name, thresholds) ->
+                    if List.length(thresholds) > 1 then Some name
+                    else None
+                )
+                |> function
+                    | [] -> Ok scenario
+                    | thresholdName::_ -> AppError.createResult(DuplicateThresholdName(scenario.ScenarioName, thresholdName))
+        )
+        |> Option.defaultValue (Ok scenario)
```

Comment:
It seems to me that it shouldn't be possible to specify duplicated thresholds, because they could contain conflicting predicates, and it would be challenging to validate this fact. As the threshold model allows to specify pretty any function as a predicate, I think, it should be sufficient here to allow just unique thresholds.

@AntyaDev Does this seem fair to you?

Reply:
We can refer to what the k6 guys added as constraints to this feature. But from my point of view, I would allow adding even conflicting predicates and just run them. Then, if the user configured something wrong, the user will see it in the result table and fix it.